src/formats/SMI.cpp

```diff
@@ -756,17 +729,23 @@ void SMIFormat::write(const Frame& frame) {
     fmt::print(*file_, "\n");
 }
 
-bool forward(TextFile& file) {
-    if (!file) {return false;}
+std::streampos SMIFormat::forward() {
+    if (!*file_) {
+        return std::streampos(-1);
+    }
 
+    auto position = file_->tellg();
     try {
-        auto line = file.readline();
+        auto line = file_->readline();
+        while (trim(line).empty()) {
```

Comment:
This is to skip blank lines?

Reply:
Yes, I had an issue without it where too many steps would be detected, so I went with this instead.

If you don't agree with it, I can revert this!

Reply:
Nope, that's fine - just confirming.